### PR TITLE
added an efficient implementation of SetMemory using DotNetCross and …

### PR DESCRIPTION
**Pull request by ogreenz**
…unfucked the Agrona Perfomance project

### CODE REVIEW

**Comment by omariom** [comment]
I would suggest using  [System.Runtime.CompilerServices.Unsafe](https://www.nuget.org/packages/System.Runtime.CompilerServices.Unsafe/) package.

**Comment by JPWatson** [comment]
Does this have any impact on performance?

**Comment by ogreenz** [comment]
It should in theory have a positive impact on performance since now the buffer are aligned. Also my latest commit handles an inconsistency with the java driver in linux and causes the aeron to fail on connect while the driver is running.

File: src/Adaptive.Agrona/Concurrent/UnsafeBuffer.cs [hunk 1]
```diff
@@ -257,13 +258,7 @@ public void SetMemory(int index, int length, byte value)
...
         {
             BoundsCheck0(index, length);
-
-            // TODO PERF Naive implementation, we should not write byte by byte, this is slow
-            //UNSAFE.SetMemory(byteArray, addressOffset + index, length, value);
-            for (var i = index; i < index + length; i++)
-            {
-                _pBuffer[i] = value;
-            }
+            Unsafe.InitBlock(((byte*)_pBuffer) + index, value, (uint)length);
```
Comment (omariom) [inline src/Adaptive.Agrona/Concurrent/UnsafeBuffer.cs]: initblk assumes that the starting address is aligned to the natural size of the machine. The operation of the initblk instructions can be altered by an immediately preceding Volatile or Unaligned prefix instruction.

Does Unsafe.InitBlock use such prefix?

Comment (ogreenz) [inline src/Adaptive.Agrona/Concurrent/UnsafeBuffer.cs]: Looking at the IL code:
```
.method public hidebysig static void  InitBlock(void* dst,
                                                  uint8 initValue,
                                                  uint32 size) cil managed aggressiveinlining
  {
    .maxstack  3
    ldarg.0
    ldarg.1
    ldarg.2
    initblk
    ret
  } // end of method Unsafe::InitBlock
```

It doesn't look like it. I wasn't aware of the initblk assumption, so either we make sure the buffer is aligned or we need to modify the IL.

**Comment by JPWatson** [comment]
I've included a couple of the changes in the latest port including the linux temp dir.
Won't merge the project.json stuff. Will be porting to new csproj format shortly (at the same time as targetting .netstandard v2.0).
Hesitant to add dependencies unless it improves performance on the hot path. 
I'm going to close this PR, let me know if there's anything I've missed.
Sorry for the delay!

**Pull request closed without merging**